qt/python/mantidqt/utils/asynchronous.py

```diff
@@ -100,6 +59,78 @@ def elapsed(start):
 
         self.finished_cb()
 
+    def abort(self):
+        """Cancel an asynchronous execution"""
+        # Implementation is based on
+        # https://stackoverflow.com/questions/5019436/python-how-to-terminate-a-blocking-thread
+        ctypes.pythonapi.PyThreadState_SetAsyncExc(ctypes.c_long(self.ident),
+                                                   ctypes.py_object(KeyboardInterrupt))
+        time.sleep(0.1)
+
+
+class SyncWithCallbackTask(AsyncTask):
```

Comment:
I think `Async` should still be in the name. The task itself is executed on a separate thread. `AsyncTaskWithBlock`? Qt has [`BlockingQueuedConnection`](https://doc.qt.io/qt-5/qt.html#ConnectionType-enum).

Reply:
Due to both yourself and @DTasev asking for changes to this, I have gone with a compromise suggested by @DTasev:
`BlockingAsyncTaskWithCallback`

Reply:
Fine with me. I think the most important part that the task is async is captured :+1: